common/src/main/java/tachyon/security/authentication/AuthenticationFactory.java

```diff
@@ -111,5 +114,27 @@ public static AuthType getAuthTypeFromConf(TachyonConf conf) {
         AuthType.NOSASL.getAuthName()));
   }
 
-  // TODO: add methods of getting different Thrift class in follow-up PR.
+  /**
+   * For server side, this method return a TTransportFactory based on the auth type. It is used as
+   * one argument to build a Thrift TServer.
+   * @return a corresponding TTransportFactory
+   * @throws SaslException if building a TransportFactory fails
```

Comment:
It also throws `UnsupportedOperationException`. (I don't remember if `UnsupportedOperationException` is also `SaslException`)